src/mapping/Partition.hpp

```diff
@@ -0,0 +1,315 @@
+#pragma once
+
+#include <Eigen/Core>
+
+#include <boost/container/flat_set.hpp>
+
+#include "mapping/RadialBasisFctSolver.hpp"
+#include "mapping/config/MappingConfiguration.hpp"
+#include "mapping/impl/BasisFunctions.hpp"
+#include "mesh/Filter.hpp"
+#include "precice/types.hpp"
+#include "utils/Event.hpp"
+
+namespace precice {
+
+namespace mapping {
+
+/**
+ * @brief Mapping using partition of unity decomposition strategies
+ */
```

Comment:
Please extend / modify. "Mapping" might also be misleading here as the class does not inherit from `Mapping`.